### PR TITLE
Update preprocessing section

### PR DESCRIPTION
**Pull request by jorainer**
This is a quite substantial PR. I've updated all sections from the preprocessing expanding/reformulating/restructuring the description. There are only few changes to the R code.

### CODE REVIEW

File: end-to-end-untargeted-metabolomics.Rmd [hunk 1]
```diff
@@ -927,122 +996,135 @@ QC_samples <- sampleData(data)$phenotype == "QC"
...
+anchor peaks defined in the subset of QC samples. QC samples, e.g. a pool of the
+study samples, repeatedly measured within an experiment are ideal for alignment,
+because the same ions are expected to be quantified in all of them enabling
+easier and more reliable definition of anchor peaks. Despite having external QCs
+in our data set, we still use the subset-based alignment assuming retention time
+shifts to be independent of the different sample matrix (human serum or plasma).
 
 ```{r}
 #' Define parameters of choice
-#' phili: do we keep the subset based alignment ?
```
Comment (jorainer) [inline end-to-end-untargeted-metabolomics.Rmd:1026]: yes, I would keep the subset-based alignment.

File: end-to-end-untargeted-metabolomics.Rmd [hunk 2]
```diff
@@ -1169,73 +1189,131 @@ eic_met <- eic_is["methionine_13C_15N"]
...
+`matchValues()` function from the `r Biocpkg("MetaboAnnotation")` package using
+the `MzRtParam` method to identify all chromatographic peaks with similar *m/z*
+(+/- 50ppm) and retention time (+/- 10 seconds) to the internal standard's
+values. We perform this matching separate for each sample and select for each
+internal standard, in each sample, using the `filterMatches()` and
+`SingleMatchParam`, the chromatographic peak with the highest intensity.
 
-Once the different chromatographic peaks have been aligned, we can proceed to the
-next step in the data pre-processing pipeline, correspondence.
+```{r}
```
Comment (jorainer) [inline end-to-end-untargeted-metabolomics.Rmd:1237]: Here I also modified the R code trying to make it a bit cleaner and splitting it into different code blocks.

File: end-to-end-untargeted-metabolomics.Rmd [hunk 3]
```diff
@@ -794,9 +818,15 @@ legend("topright", col = col_phenotype,
...
 
-These EICs seems to indicate that our settings are suitable for this dataset.
-We can proceed to apply this algorithm to the entire dataset, and then extract
-the EICs for our two test ions to confirm that the process has been successful.
+Our custom settings seem thus to be suitable for our dataset. We now proceed and
+apply them to the entire dataset, extracting EICs again for the same ions to
+evaluate and confirm that chromatographic peak detection worked as
+expected. Note that we change the value for parameter `snthresh` to its default,
+because, as mentioned above, background noise estimation is more reliable when
+performed on the full data set. Parameter `chunkSize` of `findChromPeaks()`
```
Comment (philouail) [inline end-to-end-untargeted-metabolomics.Rmd:826]: Should we relate it/mention to when parallel processing was set up at the beginning ? User might not make the link between that

Comment (jorainer) [reply end-to-end-untargeted-metabolomics.Rmd:826]: hm, difficult to say. I would like to keep this document as "general" as possible and keep the details maybe in the vignettes/tutorials? I'm just afraid we make this workflow too large.

**Pull request merged**